### ISSUE TITLE
More robust shader development workflow

### ISSUE THREAD

**Issue by acmcarther** [acmcarther/space_coop#39]
The traditional gfx way is to statically include your shader code via include_bytes!. This makes development incredibly annoying, as the compile time for the application is significant.

A current workaround is to dynamically load the shaders, but they had to be moved out of the crate directory altogether to keep cargo from recompiling anyway.

**Comment by rschifflin** [comment]
the `include_bytes!` macro compiles the shader at macro expansion time every build?

**Comment by acmcarther** [comment]
Not exactly.. It dumps the bytes into a static value. Shader compilation happens at runtime.

**Comment by acmcarther** [comment]
I've got one solution in mind that losely ties into #90. If we reduce the surface area of the shader importing code (say by injecting the dependency into the renderer), then we wont have to rebuild very much when it changes.
